runhouse/resources/secrets/provider_secrets/ssh_secret.py

```diff
@@ -134,3 +169,86 @@ def _file_to(
             system.call(key, "_write_to_file", path=path, values=values)
             remote_priv_file = file(path=path, system=system)
         return remote_priv_file
+
+    @classmethod
+    def setup_ssh_creds(cls, ssh_creds: Union[dict, str], resource_name: str):
+        """
+        this method creates an SSHSecret instance based o n the passed values. If the passed values are paths to private
+         and/or public keys, this method extracts the content of the files saved in those files, in order for them to
+         be saved in den. (Currently if we just pass a path/to/ssh/key to SSHSecret constructor, the content of the file
```

Comment:
I believe that in main currently, the contents are already saved to vault (but not to rns which is not secure).

e.g. the values for the following secret show up in vault
```
secret = rh.provider_secret(provider="ssh", name="sky", path="~/.ssh/sky-key")
secret.save()
````

Reply:
also currently `SSHSecret` is meant to represent specifically an ssh key-pair with the values dict keys corresponding to that, rather than a generic ssh cluster creds secret. adding/changing the values keys will cause the built-in  functions like `.write()` to file to break.

We should eventually build out a `SSH/ClusterCredsSecret` for this case. Another option in the meantime is to convert this to return a generic secret instead (move this function to a general utils file, and can utilize SSHSecret functions to help with loading values etc).

Reply:
Agreed. Made the ssh_cluter creds to be generic secret, and moved this function to secrets/utils.py for now.